libp2p/src/builder.rs

```diff
@@ -319,4 +318,56 @@ mod tests {
             .unwrap()
             .build();
     }
+
+    #[test]
+    #[cfg(all(feature = "tokio", feature = "tcp", feature = "tls", feature = "yamux"))]
+    fn tcp_bandwidth_logging() -> Result<(), Box<dyn std::error::Error>> {
+        let (builder, _logging) = SwarmBuilder::with_new_identity()
+            .with_tokio()
+            .with_tcp(
+                Default::default(),
+                libp2p_tls::Config::new,
+                libp2p_yamux::Config::default,
+            )?
+            .with_bandwidth_logging();
+
+        builder
+            .with_behaviour(|_| libp2p_swarm::dummy::Behaviour)
+            .unwrap()
+            .build();
```

Comment:
Can you open up a follow-up issue to discuss options of making this chainable? Loses the appeal of a builder if you can't do it in one go.

Reply:
:+1: https://github.com/libp2p/rust-libp2p/issues/4643